WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java

```diff
@@ -501,26 +501,21 @@ protected void finishCurrentActivity(final List<Map<String, Object>> userBlogLis
         if (!isAdded()) {
             return;
         }
-        getActivity().runOnUiThread(new Runnable() {
-            @Override
-            public void run() {
-                if (userBlogList != null) {
-                    if (mInhibitMagicLogin) {
-                        // just finish the login activity and return to the its "caller"
-                        getActivity().setResult(Activity.RESULT_OK);
-                        getActivity().finish();
-                    } else {
-                        // move on the the main activity
-                        Intent intent = new Intent(getActivity(), WPMainActivity.class);
-                        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
-                        intent.putExtra(SignInActivity.MAGIC_LOGIN, true);
-
-                        getActivity().startActivity(intent);
-                    }
 
-                }
+        if (userBlogList != null) {
```

Comment:
Hmmm, good point I guess this came from a merge with develop, I'll double check.